src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestRetryFailedManagedIndexAction.kt

```diff
@@ -95,55 +114,64 @@ class RestRetryFailedManagedIndexAction(
+                                            "failed to update IndexMetaData"
+                                        )
                                     })
                                 }
-                                buildInvalidIndexResponse(builder)
+                                buildInvalidIndexResponse(builder, failedIndices)
                                 return BytesRestResponse(RestStatus.OK, builder.endObject())
                             }
                         }
                     )
                 } catch (e: ClusterBlockException) {
                     failedIndices.addAll(listOfIndexMetaData.map {
-                        FailedIndex(it.first.name, it.first.uuid, "failed to update with ClusterBlockException. ${e.message}")
+                        FailedIndex(
+                            it.first.name,
+                            it.first.uuid,
+                            "failed to update with ClusterBlockException. ${e.message}"
+                        )
                     })
-                    buildInvalidIndexResponse(builder)
+                    buildInvalidIndexResponse(builder, failedIndices)
                     channel.sendResponse(BytesRestResponse(RestStatus.OK, builder.endObject()))
                 }
             } else {
-                buildInvalidIndexResponse(builder)
+                buildInvalidIndexResponse(builder, failedIndices)
                 channel.sendResponse(BytesRestResponse(RestStatus.OK, builder.endObject()))
             }
         }
 
-        private fun buildInvalidIndexResponse(builder: XContentBuilder) {
-            if (failedIndices.isNotEmpty()) {
-                builder.field(FAILURES, true)
-                builder.startArray(FAILED_INDICES)
-                for (failedIndex in failedIndices) {
-                    builder.startObject()
-                    builder.field("index_name", failedIndex.name)
-                    builder.field("index_uuid", failedIndex.uuid)
-                    builder.field("reason", failedIndex.reason)
-                    builder.endObject()
-                }
-                builder.endArray()
-            } else {
-                builder.field(FAILURES, false)
-            }
-        }
-
         private fun populateList(state: ClusterState, startState: String?) {
-            for (indexMetadataEntry in state.metaData.indices) {
-                val indexMetaData = indexMetadataEntry.value
+            for (indexMetaDataEntry in state.metaData.indices) {
+                val indexMetaData = indexMetaDataEntry.value
                 val managedIndexMetaData = indexMetaData.getManagedIndexMetaData()
                 when {
                     indexMetaData.getPolicyName() == null ->
-                        failedIndices.add(FailedIndex(indexMetaData.index.name, indexMetaData.index.uuid, "This index is not being managed."))
+                        failedIndices.add(
```

Comment:
np: same here, for me it looks like better in single line. Was this part of auto formatting from intellij?

Reply:
Changed to single line.